### ISSUE TITLE
Feature discussion: Multiple HTML files and working links

### ISSUE THREAD

**Issue by TeddyBones** [porsager/flems#28]
I didn't know how you'd feel about such a feature, so I thought it would be best to bring it up here.  Multiple HTML files is handy for teaching students how websites work.

Multiple HTML files is relatively easy to implement. The trickery comes when trying to make links work to switch between them. I have a working local branch, but it needs polishing before I can make a pull request.  

The way I've done it there, is having a new value in `state` called `htmlInView`, that is set to whichever html file that was clicked last. I've also added a script in the iframe that catches all links pointing to local files and changes them to parameter/hash values.  Then I monitor the location.href of the runtime and change `htmlInView` if it points to a html file in `state.files`. It's working, but I don't know if this breaks the purpose of a sandbox or if it's an easier way to do this?

Any thoughts?

**Comment by pygy** [comment]
Would you want the JS/CSS files to be loaded in all pages?

**Comment by TeddyBones** [comment]
Yes, for simplicity.  But when using multiple HTML, I think I'll just not make a JS file.
The same CSS in all pages is a lot less problematic.

**Comment by porsager** [comment]
I think this is an interesting idea, and it could be solved quite elegantly if https://github.com/porsager/flems/issues/6 gets implemented. A dropdown to select the active html file could be shown next to the address field if there are more than one .html file..

**Comment by TeddyBones** [comment]
I don't think #6 should get implemented, as there are a lot of people not using routing. Could be optional though. 🙂 

Do you have any suggestions on how to switch pages when overriding local links in the iFrame? I don't think my implementation is very good.

**Comment by TeddyBones** [comment]
Is there a safe way to message which page that should be active from inside the iFrame? In my prototype I used an url argument that was picked up in the location variable, but this might mess up some scripts. Is there another way you can think of or is this undoable?
